### ISSUE TITLE
Dataset and tokenization module for NT v2

### ISSUE THREAD

**Issue by smodee** [smodee/pathogen-classifier#9]
## Summary

Create `src/dataset.py` — the bridge between CSV data, the augmentation module (`src/augmentation.py`), and the Nucleotide Transformer v2 50M model.

**Priority: high | No dependencies | Can run in parallel with #8, #10**
**Resolves:** #7 (DataLoader integration test)

## Implementation

Create `src/dataset.py` with:

### Constants
```python
FAMILIES = ['Poxviridae', 'Coronaviridae', 'Paramyxoviridae', 'Filoviridae', 
            'Flaviviridae', 'Arenaviridae', 'Orthomyxoviridae']
LABEL_TO_IDX = {fam: i for i, fam in enumerate(FAMILIES)}
IDX_TO_LABEL = {i: fam for i, fam in enumerate(FAMILIES)}
```
Must match `FAMILY_ORDER` from `src/data_exploration.py`.

### Key classes
- **`SequenceDataset(csv_path)`** — PyTorch Dataset returning `{'sequence': str, 'label': int, 'id': str}`. Compatible with `GenomicAugmentationDataset` from `src/augmentation.py` (which expects items with `'sequence'` and `'label'` keys).
- **`NTCollator(tokenizer, max_length=1000)`** — collate function that tokenizes a batch of dicts into `{'input_ids', 'attention_mask', 'labels'}` tensors. NT v2 uses 6-mer tokenization with a 1000-token context window.
- **`compute_class_weights(csv_path, device) → torch.FloatTensor(7,)`** — inverse-frequency weights for addressing class imbalance (issue #6).

### Design constraint
Tokenization happens in the **collator** (after augmentation), NOT in `__getitem__`. The augmentation wrapper operates on raw sequence strings:
```
SequenceDataset → GenomicAugmentationDataset → NTCollator → tensors
```

### Model-specific notes
- NT v2 tokenizer requires `trust_remote_code=True`
- 6-mer tokenization: a 5000bp sequence → ~833 tokens (fits in 1000-token context)
- Use: `AutoTokenizer.from_pretrained("InstaDeepAI/nucleotide-transformer-v2-50m-multi-species", trust_remote_code=True)`

## Tests

Create `tests/test_dataset.py`:
- `SequenceDataset`: length, key names, label dtype/range, id format
- `NTCollator`: output tensor shapes, dtype, key names
- `compute_class_weights`: shape (7,), all positive, sums correctly
- **End-to-end integration:** `SequenceDataset → GenomicAugmentationDataset → DataLoader(collate_fn=NTCollator) → valid tensors` (resolves #7)

## Existing code to reuse
- `GenomicAugmentationDataset` from `src/augmentation.py` (items must have `'sequence'` str and `'label'` keys)
- `FAMILY_ORDER` from `src/data_exploration.py`
- CSV format: `id, parent_sequence_id, chunk_index, sequence, length, family`

## Acceptance criteria
- [ ] `SequenceDataset` loads CSV and returns correct dict format
- [ ] `NTCollator` produces valid `input_ids`, `attention_mask`, `labels` tensors
- [ ] End-to-end pipeline works: Dataset → Augmentation → DataLoader → tensors
- [ ] Class weights computed correctly
- [ ] All tests pass: `pytest tests/test_dataset.py -v`
